src/base/bittorrent/torrenthandleimpl.cpp

```diff
@@ -200,6 +200,26 @@ TorrentHandleImpl::TorrentHandleImpl(Session *session, const lt::torrent_handle
         if (filesCount() == 1)
             m_hasRootFolder = false;
     }
+
+    // TODO: Remove the following upgrade code in v.4.4
```

Comment:
Maybe it is a good idea to wrap it into a function (or just add comment about end of upgrade code)? That way we can see the boundary of the upgrade code.

Reply:
What happens to users who skip enough versions to not run the upgrade code in this case? Do the unwanted folders simply get ignored and do nothing?

Reply:
>Do the unwanted folders simply get ignored and do nothing?

Naturally, otherwise any upgrade code would have to remain forever.

Reply:
>Maybe it is a good idea to wrap it into a function (or just add comment about end of upgrade code)? That way we can see the boundary of the upgrade code.

IMO, function is overkill in this case. Wrapped by comments.

Reply:
>Do the unwanted folders simply get ignored and do nothing?

The files remain in ". unwanted" folder but will be handled as regular ones.

Reply:
> 
> 
> (unlikely anyone would skip so many

Not unlikely. The new versions introduced a couple of breaking changes such as #12446 so many people are still on 4.1.4 and older waiting for these regression bugs getting fixed first. And considering how often these regression bugs are ignored, it bubbles up.

Reply:
> unlikely anyone would skip so many
Some of us are on LTS distros, I'm using 4.1.5 and next release is in about 3 years :P